### PR TITLE
[ABW-3335] Sort seed phrases correctly

### PR DESCRIPTION
**Pull request by matiasbzurovski**
Jira ticket: [ABW-3335](https://radixdlt.atlassian.net/browse/ABW-3335?atlOrigin=eyJpIjoiYjNmYjNhMmJhZWQ1NGUxMGFkMWJlNmNiMmYzYTc3OGIiLCJwIjoiaiJ9)

## Description
This PR sorts seed phrases with desired criteria.

## Notes
Making this PR against Security Center PR since `main` currently doesn't allow accessing the `Seed Phrases` view.

[ABW-3335]: https://radixdlt.atlassian.net/browse/ABW-3335?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: RadixWallet/Features/SettingsFeature/DisplayMnemonics/Coordinator/DisplayMnemonics.swift [hunk 1]
```diff
@@ -190,3 +193,33 @@ extension DisplayEntitiesControlledByMnemonic.State {
 		self.isMnemonicMarkedAsBackedUp = true
 	}
 }
+
+// MARK: - DisplayMnemonics.ComparableEntities
+private extension DisplayMnemonics {
+	/// A helper struct to sor mnemonics using the following criteria:
```
Comment (CyonAlexRDX) [inline RadixWallet/Features/SettingsFeature/DisplayMnemonics/Coordinator/DisplayMnemonics.swift:199]: Sort, typo

File: RadixWallet/Features/SettingsFeature/DisplayMnemonics/Coordinator/DisplayMnemonics.swift [hunk 2]
```diff
@@ -190,3 +193,33 @@ extension DisplayEntitiesControlledByMnemonic.State {
...
+			let lhs = lhs.deviceFactorSource
+			let rhs = rhs.deviceFactorSource
+			if lhs.isExplicitMain {
+				return true
+			} else if rhs.isExplicitMain {
+				return false
+			} else {
+				if lhs.isBDFS, rhs.isBDFS {
+					return lhs.common.addedOn < rhs.common.addedOn
+				} else if lhs.isBDFS {
```
Comment (CyonAlexRDX) [inline RadixWallet/Features/SettingsFeature/DisplayMnemonics/Coordinator/DisplayMnemonics.swift:217]: The is last if else can be:

`else { return lhs.isBDFS }` right?

**Review by CyonAlexRDX — APPROVED** [review]
Nice!

File: RadixWallet/Features/SettingsFeature/DisplayMnemonics/Coordinator/DisplayMnemonics.swift (hunk 2 above)
Comment (matiasbzurovski) [reply RadixWallet/Features/SettingsFeature/DisplayMnemonics/Coordinator/DisplayMnemonics.swift]: 🤦 
thanks!

**Pull request merged**